pkgs/os-specific/linux/lkrg/umh-enforce-0-by-default.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/src/p_lkrg_main.c b/src/p_lkrg_main.c
+index 9ed9d37..0fa83ed 100644
+--- a/src/p_lkrg_main.c
++++ b/src/p_lkrg_main.c
+@@ -29,7 +29,7 @@ unsigned int pint_enforce = 1;
+ unsigned int pcfi_validate = 2;
+ unsigned int pcfi_enforce = 1;
+ unsigned int umh_validate = 1;
+-unsigned int umh_enforce = 1;
++unsigned int umh_enforce = 0;
```

Comment:
To be honest, I'm not a fan of this approach. 

IMO, it should be handled either by setting appropriate kernel parameters during boot or sysctl config.
Patching the module seems like a last resort when nothing else would work.

Reply:
hmmm, I have tried turning it off by cmdline and as far as I remember it didn't work(?) I'll tinker with it further tomorrow, but yeah, I can drop this

Reply:
Got an idea and I'll try generating a patch for p_umh_global on every nix evaluation so umh actually works correctly 😎

Reply:
@surfaceflinger 🤔 not sure. Then everytime somebody need recompile module 🤔  
Maybe we can write nixosModule for it and use sysctl 🤔

Reply:
@chivay, I talked to @RaitoBezarius and it seems like umh filter will be pointless and mostly problematic with Nix.

We can theoretically patch paths in src but then lkrg would depend on packages like systemd-coredump.

We have 3 options left:
- Patching out umh like I did above and instead relying on nix trustedusers + noexec for everything that isn't /nix/store
- Just bumping the version and let others have to turn off umh manually (It doesn't work correctly from end user perspective anyway, 1st option wins over this imo)
- We take this issue upstream.

If we're going upstream, I think about booting lkrg without umh, list of allowed UMHs would be reported to the interface and then we could switch umh filter on - without option to turn it off
This won't work with `nixos-rebuild switch` if some umh gets updated.

Reply:
This filtering module seems to come with a lot of problems, imagine you put in there the Nix store paths and perform an upgrade of NixOS, you need to reload the new kernel module before continuing otherwise all the new usermodehelpers will be broken.

Reply:
I guess since we're going to have secureboot in nixpkgs soon, maybe reporting umhs to the module could be done in initrd which would be signed in an UKI

Reply:
We won't have kernel lockdown before we fix the determinism of module signatures, though.